buildpack/bom.go

```diff
@@ -35,13 +35,20 @@ func (v *defaultBOMValidator) ValidateBOM(bp GroupBuildpack, bom []BOMEntry) ([]
 
 func (v *defaultBOMValidator) validateBOM(bom []BOMEntry) error {
 	if len(bom) > 0 {
-		return errors.New("bom table isn't supported in this buildpack api version. The BOM should be written to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>")
+		v.logger.Warn("BOM table is deprecated in this buildpack api version. The BOM should be written to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.")
 	}
+
+	for _, entry := range bom {
+		if entry.Version != "" {
+			return fmt.Errorf("bom entry '%s' has a top level version which is not allowed. The buildpack should instead set metadata.version", entry.Name)
+		}
+	}
```

Comment:
I'm sorry to be annoying, but is it possible to add a test for this?

Reply:
Maybe we could re-use the block here: https://github.com/buildpacks/lifecycle/blob/4a6089b4830b0012452406caed1f89c1dcf9aba8/buildpack/build_test.go#L983

Reply:
_Addressed in latest commit._